cmd/create.go

```diff
@@ -99,15 +89,50 @@ A full list of component types that can be deployed is available using: 'odo com
 			// we want to use and save absolute path for component
 			dir, err := filepath.Abs(componentLocal)
 			checkError(err, "")
-			err = component.CreateFromDir(client, componentName, componentType, dir, applicationName)
+			err = component.CreateFromDir(client, componentName, componentType, dir, applicationName, false)
 			checkError(err, "")
 			fmt.Printf("Component '%s' was created.\n", componentName)
 			fmt.Printf("To push source code to the component run 'odo push'\n")
+		} else if len(componentBinary) != 0 {
+			fi, err := os.Stat(componentBinary)
+			if err != nil {
+				checkError(err, "")
+			}
+			if fi.IsDir() {
+				fmt.Println("Please provide path to binary instead of a directory")
+				os.Exit(1)
+			}
+
+			err = component.CreateFromDir(client, componentName, componentType, componentBinary, applicationName, true)
+			checkError(err, "")
+
+			sourceType, sourcePath, err := component.GetComponentSource(client, componentName, applicationName, projectName)
```

Comment:
Why are we getting ComponentSource right after we create it?

Reply:
@kadel I wanted to be extra sure about the source but now I have realized that I am checking at the start itself, will fix this